File: src/main/java/um/haberes/rest/model/view/AsignadoClase.java
/**
 * 
 */
package um.haberes.rest.model.view;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.Table;

import org.hibernate.annotations.Immutable;

import um.haberes.rest.model.view.pk.AsignadoClasePk;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Immutable
@Table(name = "vw_asignado_clase")
@IdClass(AsignadoClasePk.class)
@NoArgsConstructor
@AllArgsConstructor
public class AsignadoClase implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 9025541214131527308L;

	@Id
	private Long legajoId;
	
	@Id
	private Integer dependenciaId;

	@Id
	private Long cargoClaseId;

	private Long periodoDesde;
	private Long periodoHasta;
	private BigDecimal basicoDesde = BigDecimal.ZERO;
	private BigDecimal basicoHasta = BigDecimal.ZERO;

}

File: src/main/java/um/haberes/core/kotlin/model/view/AsignadoClase.kt
package um.haberes.core.kotlin.model.view

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import org.hibernate.annotations.Immutable
import java.math.BigDecimal

@Entity
@Table(
    name = "vw_asignado_clase",
    uniqueConstraints = [UniqueConstraint(columnNames = ["legajoId", "dependenciaId", "cargoClaseId"])]
)
@Immutable
data class AsignadoClase(

    @Id
    var uniqueId: String? = null,
    var legajoId: Long? = null,
    var dependenciaId: Int? = null,
    var cargoClaseId: Long? = null,
    var periodoDesde: Long? = null,
    var periodoHasta: Long? = null,
    var basicoDesde: BigDecimal = BigDecimal.ZERO,
    var basicoHasta: BigDecimal = BigDecimal.ZERO

)
